var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -298,13 +299,5 @@ def test(self):
             exe = os.path.join(self.prefix.bin, test[0])
             args = test[1:]
             reason = 'test {0}: "{1}"'.format(i, " ".join(test))
-            self.run_test(
-                exe,
-                args,
-                [],
-                0,
-                installed=False,
-                purpose=reason,
-                skip_missing=True,
```

Comment:
Based on this option, my inclination would be to raise `SkipTest` at the top of the test methods if the path to the executable does not exist.  (See https://docs.python.org/3/library/os.path.html for the `exists` method.)

Reply:
This request still applies for both of the test methods.

You could even consider a refactor similar to what you did in one of your earlier PRs.

Reply:
@tldahlgren, is it possible to simply call `ctest`? That would take care of which tests should be run for which build configuration. This looks like duplicating in Spack what has been already implemented in CMake.

Reply:
Definitely possible.  There was a recent PR that was updated to the new test API that does just that.  It's `vtk-m` (#44705).

Reply:
Yes, it can be done.  There would be one stand-alone test method (not called `test`) that could set up, build, and run the tests.  

A recently updated example is the `vtk-m` package.  Looks like the test output didn't get added as a comment to #44705 but you could `spack -v test run vtk-m` (if or once you install it) to see what the test output would look like if you're interested.